gap/ColouredComplexes/isoscelesColouring.gi

```diff
@@ -0,0 +1,777 @@
+    colours := List(edges, edg -> List(edg, e -> ColoursOfEdges(colComp)[e]));
+        # check that each face has edges of 2 colours and that
+        # these are the same for all faces
+    for c in colours do Sort(c); od;
+    c := __SIMPLICIAL_BoundPositions(colours);
+    for i in c do
+        if colours[i] <> colours[c[1]] then return false; fi;
+    od;               
+    return true;
+    end
+);
+
+
+InstallOtherMethod( IsIsoscelesColouredSurface,
+    "for an object", [IsObject], function(obj)
+        if IsEdgeColouredPolygonalComplex(obj) then
+            TryNextMethod();
+        else
+            return false;
+        fi;
+    end
+);
+#######################################
+##
+##      ColouredEdgesOfFaces
+##
+InstallMethod( ColouredEdgesOfFaceNC,
+    "for an edge two-coloured polygonal complex and a face",
+    [IsEdgeColouredPolygonalComplex and IsEdgeTwoColouring, IsPosInt],
+    function(rbComp, face)
+        return ColouredEdgesOfFaces(rbComp)[face];
+    end
+);
+    RedispatchOnCondition( ColouredEdgesOfFaceNC, true, 
+        [IsEdgeColouredPolygonalComplex, IsPosInt], 
+        [IsEdgeTwoColouring,], 0 );
+
+InstallMethod( ColouredEdgesOfFace,
+    "for an edge two-coloured polygonal complex and a face",
+    [IsEdgeColouredPolygonalComplex and IsEdgeTwoColouring, IsPosInt],
+    function(rbComp, face)
+        __SIMPLICIAL_CheckFace( PolygonalComplex(rbComp), face, "ColouredEdgesOfFace" );
+        return ColouredEdgesOfFaceNC(rbComp,face);
+    end
+);
+    RedispatchOnCondition( ColouredEdgesOfFace, true, 
+        [IsEdgeColouredPolygonalComplex,IsPosInt], 
+        [IsEdgeTwoColouring,], 0 );
+
+InstallMethod( ColouredEdgesOfFaces,
+    "for an edge two-coloured polygonal complex",
+    [IsEdgeColouredPolygonalComplex and IsEdgeTwoColouring],
+    function(rbComp)
+        local edges, colEdgesOfFaces, face, edgesOfCol, e;
+
+        colEdgesOfFaces := [];
+        for face in Faces( PolygonalComplex(rbComp) ) do
+            edges := EdgesOfFaces(PolygonalComplex(rbComp))[face];
+            edgesOfCol := [[],[]];
+            for e in edges do
+                Add(edgesOfCol[ ColoursOfEdges(rbComp)[e] ], e);
+            od;
+            colEdgesOfFaces[face] := edgesOfCol;
+        od;
+                
+        return colEdgesOfFaces;
+    end
+);
+    RedispatchOnCondition( ColouredEdgesOfFaces, true, 
+        [IsEdgeColouredPolygonalComplex], 
+        [IsEdgeTwoColouring], 0 );
+
+
+
+InstallMethod( ApexVertexOfFace,
+    [IsIsoscelesColouredSurface, IsPosInt],
+    function(surf, face)
+
+     local coledges, v;
+               
+
+     __SIMPLICIAL_CheckFace(surf, face, "ApexVertexOfFace");
+     coledges := ColouredEdgesOfFace( surf, face);
+     if Length(coledges[1]) = 2 then
+        coledges := coledges[1];
+     elif Length(coledges[2]) = 2 then
+        coledges := coledges[2];
+    else Error("ApexVertexOfFace: Not an isosceles-coloured face");
```

Comment:
This error can never be raised: The method can only be called for IsIsoscelesColouredSurfaces